fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Deserializable.kt

```diff
@@ -74,9 +74,17 @@ private fun <T : Any, U : Deserializable<T>> Request.response(deserializable: U,
     return this
 }
 
-fun <T : Any, U : Deserializable<T>> Request.response(deserializable: U): Triple<Request, Response, Result<T, FuelError>> = try {
-    val response = taskRequest.call()
-    Triple(this, response, Result.Success(deserializable.deserialize(response)))
-} catch (error: FuelError) {
-    Triple(this, error.response, Result.error(error))
+fun <T : Any, U : Deserializable<T>> Request.response(deserializable: U): Triple<Request, Response, Result<T, FuelError>> {
+    var response : Response?  = null
+    val result = Result.of<T,FuelError> {
+        response = taskRequest.call()
+        deserializable.deserialize(response!!)
+    }
+    return Triple(this, response ?: result.component2()?.response!!, result)
 }
+
+fun <V : Any>Result.Companion.ofCatching(c : (exception:Exception) -> FuelError,f: () -> V): Result<V, FuelError> = try {
```

Comment:
Hmm I don't see this being used anyware.

Reply:
I don't know why this is not showing as modified but it is

Reply:
Can you tell me why the regular of is not suitable for this?

Reply:
Because when an error is thrown in deserialisation it will is likely to be as classCastExecption, which cannot be cast to a Fuel error, hence the exception is not throw. you need to pass in a lambda to describe how you change map all exception to the chosen error type,

I have raised a pr in Result, I will update it now with a failing test to demonstrate what I mean and you can then check out that branch?

more detail can be found here https://github.com/kittinunf/Result/pull/39

Reply:
We can't just replace 

> catch (error: FuelError)

with 
```kotlin
catch (ex: Exception)
``` 
https://github.com/kittinunf/Fuel/pull/381/files#diff-fececc99239d4e844c4cce8342aa7668L80

and forward it as 
```kotlin
Result.error(FuelError(error)))
```

?

Reply:
Maybe catch Exception in TaskRequest.call and wrap it as FuelError, and then catch it again in Request.response and wrap it again?